drivers/d3d12/SCsub

```diff
@@ -0,0 +1,204 @@
+mesa_private_inc_paths = [v for v in mesa_private_inc_paths if v not in mesa_blacklist_inc_paths]
+# Prepending these to avoid some conflicts, at least with OS GL headers.
+env_thirdparty_mesa.Prepend(CPPPATH=mesa_private_inc_paths)
+
+if not env["vulkan"]:
+    # Some Mesa file needs to use Vulkan includes.
+    env_thirdparty_mesa.Append(CPPPATH="#thirdparty/vulkan/include")
+
+mesa_sources += [v for v in list(Path(mesa_absdir).rglob("*.c"))]
+mesa_sources += [v for v in list(Path(mesa_absdir).rglob("*.cpp"))]
+mesa_sources = [str(v).replace(mesa_absdir, mesa_dir) for v in mesa_sources]
+mesa_sources = [v.replace("\\", "/") for v in mesa_sources]
+# Avoid build results depending on if generated files already exist.
+mesa_sources = [v for v in mesa_sources if not v.startswith(mesa_gen_dir)]
+mesa_sources.sort()
+# Include the list of the generated ones now.
+mesa_sources += mesa_gen_sources
+env_thirdparty_mesa.add_source_files(thirdparty_obj, mesa_sources)
+
+# Clean C/C++/both flags to avoid std clashes.
+for key in ["CFLAGS", "CXXFLAGS", "CCFLAGS"]:
+    env_thirdparty_mesa[key] = [v for v in env_thirdparty_mesa[key] if "/std:" not in v and "-std=" not in v]
+
+# Added by ourselves.
+extra_defines = [
+    "WINDOWS_NO_FUTEX",
+]
+
+# These defines are inspired by the Meson build scripts in the original repo.
+extra_defines += [
+    "__STDC_CONSTANT_MACROS",
+    "__STDC_FORMAT_MACROS",
+    "__STDC_LIMIT_MACROS",
+    ("PACKAGE_VERSION", '\\"' + Path(mesa_absdir + "/VERSION").read_text().strip() + '\\"'),
+    ("PACKAGE_BUGREPORT", '\\"https://gitlab.freedesktop.org/mesa/mesa/-/issues\\"'),
+    "PIPE_SUBSYSTEM_WINDOWS_USER",
+    "_USE_MATH_DEFINES",
+    ("_Static_assert", "static_assert"),
+]
+
+if env.msvc:
+    extra_defines += [
+        "_USE_MATH_DEFINES",
+        "VC_EXTRALEAN",
+        "_CRT_SECURE_NO_WARNINGS",
+        "_CRT_SECURE_NO_DEPRECATE",
+        "_SCL_SECURE_NO_WARNINGS",
+        "_SCL_SECURE_NO_DEPRECATE",
+        "_ALLOW_KEYWORD_MACROS",
+        ("_HAS_EXCEPTIONS", 0),
+        "NOMINMAX",
+        "HAVE_STRUCT_TIMESPEC",
+    ]
+    env_thirdparty_mesa.Append(CFLAGS=["/std:c11"])
+    env_thirdparty_mesa.Append(CXXFLAGS=["/std:c++17"])
+else:
+    env_thirdparty_mesa.Append(
+        CPPDEFINES=[
+            ("__MSVCRT_VERSION__", 0x0700),
+        ]
+    )
+    env_thirdparty_mesa.Append(CFLAGS=["-std=c11"])
+    env_thirdparty_mesa.Append(CXXFLAGS=["-std=cpp++17"])
+
+env_thirdparty_mesa.Append(CPPDEFINES=extra_defines)
+
+# This is needed since rendering_device_d3d12.cpp needs to include some Mesa internals.
+env_d3d12_rd.Prepend(CPPPATH=mesa_private_inc_paths)
+# For the same reason as above, the defines must be the same as in the 3rd-party code itself.
+env_d3d12_rd.Append(CPPDEFINES=extra_defines)
+
+# No point in fighting warnings in Mesa.
+if env.msvc:
+    env_thirdparty_mesa.Append(CCFLAGS=["/W0"])
+else:
+    env_thirdparty_mesa.Append(CCFLAGS=["-w"])
```

Comment:
```suggestion
env_thirdparty_mesa.disable_warnings()
```
Fixes MSVC warning spam:
https://github.com/godotengine/godot/blob/7df393312fa8eab98231f2c335366a1ef6205367/methods.py#L118-L128

Reply:
Awesome! Will come in next push.